buildprocess/configureWebpack.js

```diff
@@ -194,6 +194,8 @@ function configureWebpack({
       typescript: {
         memoryLimit: 4096,
         configFile: path.resolve(__dirname, "..", "tsconfig.json"),
+        build: true,
```

Comment:
Wondering why do we need `build: true` here?

Reply:
The ending parenthesis of the description of `mode` in this [table](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/blob/main/README.md#typescript-options) states it's required.

Reply:
I missed that part, but testing this further I couldn't get any consistent benchmark results. My benchmarks are showing that typecheck is slower in dev mode with `build: true`, and AFAIK we don't need tsc build output, so probably we don't need build specified. Besides that there are couple of issues in the fork-ts-checker claiming that it is not necessary to use build. 
I would go without build specified (maybe even only `write-tsbuildinfo` for mode as that is only file we write to disk), what do you think @na9da? We can always revert this part and leave it only for node if issues arouse

Reply:
I had a look at the fork-ts-checker-webpack-plugin tickets and unrelated to incremental, I found https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/issues/855 which is about TypeScript 5.6+ (which we are using). I'm all for enabling incremental mode for fork-ts-checker-webpack-plugin, but I feel the scope of this PR has increased quite a bit since we're now into reported bugs of the fork-ts-checker-webpack-plugin.

This PR gives a 9x performance improvement for tsc according to @na9da's quick measurement, and I suspect there's a bit more to be had from #7425. Could we sort out the `tsconfig.json` settings we want and get those parts merged first, and then work on getting those settings to take effect for fork-ts-checker-webpack-plugin?